internal/provider/resource.go

```diff
@@ -5,47 +5,65 @@ import (
 )
 
-func nullResource() *schema.Resource {
-	return &schema.Resource{
-		Description: `The ` + "`null_resource`" + ` resource implements the standard resource lifecycle but takes no further action.
+var (
+	_ resource.Resource = (*nullResource)(nil)
+)
 
-The ` + "`triggers`" + ` argument allows specifying an arbitrary set of values that, when changed, will cause the resource to be replaced.`,
+func NewNullResource() resource.Resource {
+	return &nullResource{}
+}
 
-		CreateContext: resourceCreate,
-		ReadContext:   resourceRead,
-		DeleteContext: resourceDelete,
+type nullResource struct{}
 
-		Schema: map[string]*schema.Schema{
+func (n nullResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_resource"
+}
+
+func (n nullResource) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `The ` + "`null_resource`" + ` resource implements the standard resource lifecycle but takes no further action.
+
+The ` + "`triggers`" + ` argument allows specifying an arbitrary set of values that, when changed, will cause the resource to be replaced.`,
+		Attributes: map[string]tfsdk.Attribute{
 			"triggers": {
 				Description: "A map of arbitrary strings that, when changed, will force the null resource to be replaced, re-running any associated provisioners.",
-				Type:        schema.TypeMap,
+				Type:        types.MapType{ElemType: types.StringType},
 				Optional:    true,
-				ForceNew:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.RequiresReplace(),
+				},
 			},
 
 			"id": {
 				Description: "This is set to a random value at create time.",
 				Computed:    true,
-				Type:        schema.TypeString,
+				Type:        types.StringType,
 			},
 		},
-	}
+	}, nil
+}
+
+func (n nullResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	diags := resp.State.SetAttribute(ctx, path.Root("id"), fmt.Sprintf("%d", rand.Int()))
+	resp.Diagnostics.Append(diags...)
 }
 
-func resourceCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	d.SetId(fmt.Sprintf("%d", rand.Int()))
-	return nil
+func (n nullResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+
 }
 
-func resourceRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	return nil
+func (n nullResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
 }
 
-func resourceDelete(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	d.SetId("")
-	return nil
+func (n nullResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	diags := resp.State.SetAttribute(ctx, path.Root("id"), "")
+	resp.Diagnostics.Append(diags...)
```

Comment:
In terraform-plugin-framework, the `id` attribute is no longer a special "resource exists" signal. If providers need to signal the resource is being removed, `resp.State.RemoveResource()` can be called, however the framework will automatically do this when `Delete` is called and there are no errors returned.

References:

- https://www.terraform.io/plugin/framework/migrating/resources/crud#migration-notes
- https://www.terraform.io/plugin/framework/resources#delete

```suggestion
```